### PR TITLE
Revised framework and headers

### PR DESCRIPTION
**Pull request by Sezzary**
## todo

- [x] I have added a changelog entry to CHANGELOG.md file on the branch/fork (if it is an internal change then it is not needed)
- [x] Pull request meets the Coding Conventions standards:https://github.com/MontyTRC89/TombEngine/blob/master/CONTRIBUTING.md#coding-conventions

## Links to issue(s) this pull request concerns (if applicable)

n/a

## Description of pull request

Update to engine foundations. Very big but this should be done at some point. It can sit around until we are comfortable merging.

- Updates `Framework.h`.
- Makes it no longer necessary to include `Framework.h` in every file manually as project settings force include it.
- Adds TEN math to the framework, meaning it is no longer necessary to include it manually everywhere. This was something I should have done initially.
- Updates headers across the entire project.
- Integrates VCpkg for easier package management. Library code no longer needs to be a part of the project.

Parts are taken from this closed PR by Tony: https://github.com/MontyTRC89/TombEngine/pull/1126
VCpkg integration by Tokyo.

### CODE REVIEW

File: TombEngine/Math/Geometry.h [hunk 1]
```diff
@@ -1,9 +1,9 @@
 #pragma once
 
-class AxisAngle;
-class EulerAngles;
-class Pose;
-class Vector3i;
+namespace TEN::Math { class AxisAngle; };
+namespace TEN::Math { class EulerAngles; };
+namespace TEN::Math { class Pose; };
+namespace TEN::Math { class Vector3i; };
```
Comment (Adngel) [inline TombEngine/Math/Geometry.h:7]: Shall it repeated namespaces or combine all under a single one?

```c++
namespace TEN::Math
{
    class AxisAngle; 
    class EulerAngles;
    class Pose;
    class Vector3i; 
};
```

**Review by Adngel — COMMENTED** [review]
There are a couple of conflicts with develop. But apart of that, I didn't saw any issue on the code, and after testing it, I neither had any issue compilation or the game running.

File: TombEngine/Math/Geometry.h (hunk 1 above)
Comment (Sezzary) [reply TombEngine/Math/Geometry.h:7]: Probably not, it might be repetitive but if there are multiple namespaces headers get messy.

**Comment by TokyoSU** [comment]
Compile fine, everything work :)

**Pull request merged**